src/fetch.rs

```diff
@@ -48,191 +47,144 @@ impl Method {
     }
 }
 
-/// Higher-level wrapper for web_sys::RequestInit.
-/// https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.RequestInit.html#method.mode
-#[derive(Clone)]
-pub struct RequestOpts {
-    // todo: Macro for this?
-    pub payload: Option<String>,
-    pub headers: HashMap<String, String>,
-    pub credentials: HashMap<String, String>,
-    pub mode: web_sys::RequestMode,
+
+#[derive(Debug)]
+pub struct Request<'a> {
+    url: &'a str,
+    init: web_sys::RequestInit,
+    headers: web_sys::Headers,
 }
 
-impl RequestOpts {
-    pub fn new() -> Self {
+impl<'a> Request<'a> {
+    pub fn new(url: &'a str) -> Self {
         Self {
-            payload: None,
-            headers: HashMap::new(),
-            credentials: HashMap::new(),
-            // https://rustwasm.github.io/wasm-bindgen/api/web_sys/enum.RequestMode.html
-            mode: web_sys::RequestMode::Cors,
+            url,
+            init: web_sys::RequestInit::new(),
+            headers: web_sys::Headers::new().expect("Error with creating Headers"),
         }
     }
-}
 
+    #[inline]
```

Comment:
This is more a question than a review. Do we really need all these #[inline]? Will it eventually be optimized away if we `inline` them?

Reply:
I'm not an expert on low-level optimizations, but my understanding is: in LTO mode Rust + LLVM will do a lot of inlining automatically, so the `#[inline]` is ***not*** necessary.

But in non-LTO mode the `#[inline]` can help with performance.

So I just got into the habit of always adding `#[inline]` to small functions which should be inlined, to help guarantee that they are inlined.

Reply:
So, I think in a wasm framework like this, performance should be focus on the DOM render/update. In other part of the framework, I think that we should focus optimizing on size instead of performance?

Reply:
Inlining (when done properly) helps with both performance and code size, since all of the function wrappers and abstractions melt away, and it enables better optimizations by the compiler.

This is Rust, we don't need to make compromises: we can make *everything* fast, including the DOM, but also including everything else too.